### PR TITLE
Match cuckoo cpp impl with py 177097602

### PR DESCRIPTION
**Pull request by ArrowAcrobatics**
This PR adds cuckoo filter to crownstone core in the util folder.
(Required core feature as both ble and uart will need it.)

Tests currently have to be executed manually. Maybe we could formalize tests a bit further through usage of a pytest/tox test stack. (That would run out of scope for this task.)

https://www.pivotaltracker.com/story/show/177097602

### CODE REVIEW

File: crownstone_core/util/cuckoofilter.py [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+
+    def size(self):
+        """
+        returns: int
+        """
+        return CuckooFilter.getsize(self.bucket_count, self.nests_per_bucket)
+
+
+if __name__ == "__main__":
+    f = CuckooFilter(50, 4)
```
Comment (AlexDM0) [inline crownstone_core/util/cuckoofilter.py]: this should be moved to a test file

File: crownstone_core/util/randomgenerator.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+    def __call__(self):
+        """
+        Shorthand for self.rand()
+        """
+        return self.rand()
+
+
+if __name__ == "__main__":
+    rand = RandomGenerator(0x1234567890abcdef)
```
Comment (AlexDM0) [inline crownstone_core/util/randomgenerator.py]: this should be moved to a test file

**Review by AlexDM0 — COMMENTED** [review]
Making tests with tox is really easy. Wouldnt it be worth it to just convert the tests to adhere to it? Otherwise this is unlikely to be done any time soon.

Tox is only used as an automator at the moment. The tests are done by executing pytest.
https://docs.pytest.org/en/stable/getting-started.html#run-multiple-tests

The only condition is that the test files start with the filename test_* and that the functions in those files follow this naming convention:
```
def test_*()
```

test_filter is a fine name for instance.

Use assert() to test things.

See the available test file for inspiration.

File: crownstone_core/util/cuckoofilter.py (hunk 1 above)
Comment (ArrowAcrobatics) [reply crownstone_core/util/cuckoofilter.py]: Check. See d456070150b93bdd18d23e8187977b6b10de7c85

File: crownstone_core/util/randomgenerator.py (hunk 2 above)
Comment (ArrowAcrobatics) [reply crownstone_core/util/randomgenerator.py]: Check, see: 5b7bbdb33de06f2a6837996df1c3e13f55565d92

**Pull request merged**